config/config_test.go

```diff
@@ -5,137 +5,106 @@ package config_test
-	cfg.RegisterComponent(config.PinTracker, maptrackerCfg)
-	return cfg
+type mockCfg struct {
+	config.Saver
 }
 
-func TestManager_ToJSON(t *testing.T) {
-	var want1 bytes.Buffer
-	json.Compact(&want1, []byte(`{
-"api": {
-"ipfsproxy": {
-"listen_multiaddress": "/ip4/127.0.0.1/tcp/9095",
-"node_multiaddress": "/ip4/127.0.0.1/tcp/5001",
-"read_timeout": "0s",
-"read_header_timeout": "5s",
-"write_timeout": "0s",
-"idle_timeout": "1m0s",
-"extract_headers_path": "/api/v0/version",
-"extract_headers_ttl": "5m0s"
-},
-"restapi": {
-"http_listen_multiaddress": "/ip4/127.0.0.1/tcp/9094",
-"read_timeout": "0s",
-"read_header_timeout": "5s",
-"write_timeout": "0s",
-"idle_timeout": "2m0s",
-"basic_auth_credentials": null,
-"headers": {},
-"cors_allowed_origins": [
-"*"
-],
-"cors_allowed_methods": [
-"GET"
-],
-"cors_allowed_headers": [],
-"cors_exposed_headers": [
-"Content-Type",
-"X-Stream-Output",
-"X-Chunked-Output",
-"X-Content-Length"
-],
-"cors_allow_credentials": true,
-"cors_max_age": "0s"
+func (m *mockCfg) ConfigKey() string {
+	return "mock"
 }
-},
-"pin_tracker": {
-"maptracker": {
-"max_pin_queue_size": 50000,
-"concurrent_pins": 10
+
+func (m *mockCfg) LoadJSON([]byte) error {
+	return nil
 }
+
+func (m *mockCfg) ToJSON() ([]byte, error) {
+	return []byte(`{"a":"b"}`), nil
 }
-}`))
-	tests := []struct {
-		name    string
-		want    []byte
-		wantErr bool
-	}{
-		{
-			"basic default config",
-			want1.Bytes(),
-			false,
-		},
-	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			cfgMgr := setupConfigManager()
-			err := cfgMgr.Default()
-			if err != nil {
-				t.Fatal(err)
-			}
-			got, err := cfgMgr.ToJSON()
-			if (err != nil) != tt.wantErr {
-				t.Errorf("Manager.ToJSON() error = %v, wantErr %v", err, tt.wantErr)
-				return
-			}
 
-			// Extract non-cluster config sections as cluster section just with each run.
+func (m *mockCfg) Default() error {
+	return nil
+}
 
-			// api section
-			gotapi, err := extractJsonSection("api", got)
-			if err != nil {
-				t.Fatal(err)
-			}
-			wantapi, err := extractJsonSection("api", tt.want)
-			if err != nil {
-				t.Fatal(err)
-			}
-			if !reflect.DeepEqual(gotapi, wantapi) {
-				t.Errorf("api section not the same: got: %s want: %s", gotapi, wantapi)
-			}
+func (m *mockCfg) Validate() error {
+	return nil
+}
 
-			// pintracker section
-			gotpintracker, err := extractJsonSection("pin_tracker", got)
-			if err != nil {
-				t.Fatal(err)
-			}
-			wantpintracker, err := extractJsonSection("pin_tracker", tt.want)
-			if err != nil {
-				t.Fatal(err)
-			}
-			if !reflect.DeepEqual(gotpintracker, wantpintracker) {
-				t.Errorf("api section not the same: got: %s want: %s", gotpintracker, wantpintracker)
-			}
-		})
-	}
+func setupConfigManager() *config.Manager {
+	cfg := config.NewManager()
+	mockCfg := &mockCfg{}
+	cfg.RegisterComponent(config.Cluster, mockCfg)
+	cfg.RegisterComponent(config.API, mockCfg)
+	cfg.RegisterComponent(config.API, mockCfg)
```

Comment:
Is there a reason there are dups lines?

You can do a `for _, sec := range SectionTypes() { cfg.RegisterComponent(sec, mockCfg) }`

Reply:
For all section except the `config.Cluster` one, that is.

Reply:
fixed myself

Reply:
@hsanjuan thanks for the fix, I had just copied it across from `cmd/ipfs-cluster-service/config.go`.